core/kernel/panic.c

```diff
@@ -29,12 +30,23 @@
 #include <kernel/thread.h>
 #include <trace.h>
 
-void __panic(const char *file __maybe_unused, int line __maybe_unused,
-		const char *func __maybe_unused)
+void __do_panic(const char *file __maybe_unused,
+		const int line __maybe_unused,
+		const char *func __maybe_unused,
+		const char *msg __maybe_unused)
 {
-	uint32_t __unused exceptions = thread_mask_exceptions(THREAD_EXCP_ALL);
+	(void)thread_mask_exceptions(THREAD_EXCP_ALL);
 
-	EMSG_RAW("PANIC: %s %s:%d\n", func, file, line);
+#if defined(CFG_TEE_CORE_DEBUG) && CFG_TEE_CORE_DEBUG != 0
+	if (msg)
```

Comment:
issue in both `EMSG_RAW()` below.

if `panic();` is called from a macro, not a C function, `__func__` will be `NULL` and panic trace will itself panic.
See thread.c canaries where `panic("...");` is called from the `DEAD_CANARY()` macro.